lib/smack/Regions.cpp

```diff
@@ -347,9 +347,6 @@ void Regions::visitCallInst(CallInst &I) {
   Function *F = I.getCalledFunction();
   std::string name = F && F->hasName() ? F->getName().str() : "";
 
-  if (I.getType()->isPointerTy())
```

Comment:
I could not wrap my head around why we needed this in the first place. So it made sense to me to remove it. Do you maybe remember why this region creating is here?

Reply:
I’m still worried about this line… making such a drastic change when we don’t understand why. And if it’s not specifically about malloc, then it seems orthogonal to the spirit of this PR.

Can’t we just guard it for the case of malloc?

Reply:
This if condition was added 4 years ago, and regions implementation changed a lot since then. I don't know why it was added, the only explanation I have is that is might be related to the implementation of contracts, which I am not familiar with. The fact that this change does not break any SVCOMP benchmarks either also makes me thing that maybe this is related to contracts.

Anyhow, are you ok if I guard this with `if (f->isDeclaration() && !EXTERNAL_PROC_IGNORE.match(name))`, which is used in the other part of the code I changed?

Reply:
Yes, for this PR I think your guard is a good idea.

I’m not against getting rid of this line completely, eventually, once we understand it a bit more, i.e., in the context of a separate issue & PR.

Reply:
Ok, I put this back in. Could you please wrap up the review (if you are happy with the change), and I'll merge it?